### ISSUE TITLE
Re: theses pdf files and location issue

### ISSUE THREAD

**Issue by mbarnett** [ualbertalib/jupiter#1853]
Submitted via email from:Era.Helpdesk@ualberta.ca

A few points:

-This is a known issue for thesis PDFs.

-Abandoned projects relating to preservation including the Grand Audit and
the Single Source of Truth were meant to help, but I do not believe
addressed this specific problem.

-In meeting with Weiwei, what was apparent is that we did not have a role
defined for auditing ERA content for same-problem-records when we
identified a record deficiency. That's why this problem is a known issue
and is not solved. We have another systemic data corruption problem with
zipfiles with videos in them. Again, abandoned work.

-I'm not sure if this just affects ProQuest PDFs but we run into this often
enough that IMO we should consider whether to re-ingest the ProQuest theses
files. *scream* I know, but we've got to narrow down this problem. If we
could at least rule it out for ProQuest theses, we could possibly conduct
an item by item check for the post 2009 e-submisssion world. Or extract the
items from 2009-the present for that time period and perhaps try to run a
script that tests whether they open or not. That would be much less to bite
off and chew than checking tens of thousands of theses manually. It would
still be a very large number so anything we could automate would be helpful.

Agree should be discussed at future group meeting.

LV

On Fri, Sep 4, 2020 at 12:53 PM Sharon Farnel <sharon.farnel@ualberta.ca>
wrote:

> Hi all,
>
> Confirmed that we do have these on dvd. I can access them next time I'm in
> the office but will wait on the discussion at DI Team Leads before heading
> in to do so as it seems better to wait until we may have a comprehensive
> list. Unless someone needs this asap in which case I'm happy to grab it.
>
> Cheers,
> Sharon
>
> On Fri, Sep 4, 2020 at 7:17 AM Sharon Farnel <sharon.farnel@ualberta.ca>
> wrote:
>
>> Thanks very much Matt.
>>
>> This explanation is very helpful. I agree; very good topic for the next
>> DI Team Leads meeting.
>>
>> I will add to the agenda and flag you and I (if you don't mind) as
>> bringing it forward so we can give this background and context.
>>
>> In the meantime, I will look to see if we do indeed have the files for
>> these items and we can get them replaced in ERA.
>>
>> Cheers,
>> Sharon
>>
>>
>>
>> On Fri, Sep 4, 2020 at 12:05 AM Matthew Barnett <mbarnett@ualberta.ca>
>> wrote:
>>
>>> Unfortunately no, there isn’t. Because these are the files that were
>>> originally ingested into Jupiter, there’s no file checksum discrepancy or
>>> similar to help us find these. And the characterization data we have
>>> doesn’t really flag mismatches of this sort.
>>>
>>> What will likely need to happen is something like an audit of all of the
>>> current checksums against records of previous checksums from older
>>> migration(s), which may turn up some mismatches. But depending on how far
>>> back these errors were misidentified as the files that were meant to be
>>> ingested, that may not be enough; there will probably need to be a thorough
>>> audit and validation of each and every file in ERA to turn up any other
>>> major content mismatches.
>>>
>>> Piyapong was working on a general high-level checksum comparison last
>>> year, and theoretically at some point relatively soon he should be freed up
>>> from the lingering issues with the Dataverse migration to return to looking
>>> at the hydranorth migration data. But if I understand conversations with
>>> Weiwei correctly, the broader ongoing responsibility for this kind of audit
>>> and validation project is intended to end up falling under Kenton’s area,
>>> so he’ll likely need to take the lead on this overall.
>>>
>>> Accordingly, we should probably surface this for discussion at the next
>>> DI Team Leads meeting?
>>>
>>> On Thursday, September 3, 2020, Sharon Farnel <sharon.farnel@ualberta.ca>
>>> wrote:
>>>
>>>> Huh. Well that explains it!
>>>>
>>>> Is there an easy way to get a list of ERA items with this issue?
>>>>
>>>> I can then confirm that we have the originals and can snag them from
>>>> the dvds.
>>>>
>>>> Cheers,
>>>> Sharon
>>>>
>>>> On Thu, Sep 3, 2020 at 4:41 PM Matthew Barnett <mbarnett@ualberta.ca>
>>>> wrote:
>>>>
>>>>> Huh. It looks like these were migrated into ERA *thinking* that we had
>>>>> the PDFs for these items, when what was actually migrated was an error
>>>>> message from Chelsea presumably from when whatever migration process
>>>>> attempted to download the files for migration. It’s not clear if this
>>>>> happened during the Hydra->Jupiter migration or sometime prior;
>>>>> regardless, someone will have to get the original copies of these files and
>>>>> upload them to these items.
>>>>>
>>>>> ---
>>>>> Matthew Barnett
>>>>>
>>>>> Senior Programmer Analyst
>>>>> University of Alberta Library
>>>>> 2-10 Cameron Library
>>>>> Edmonton, Alberta, Canada T6G 2J8
>>>>> Email: mbarnett@ualberta.ca <mbarnett@ualberta.ca> (he/him)
>>>>>
>>>>> On Sep 3, 2020, at 4:34 PM, Mariana P. Olea <paredeso@ualberta.ca>
>>>>> wrote:
>>>>>
>>>>> Hi devs,
>>>>>
>>>>> I stumbled upon a couple theses for which I couldn' access pdfs using
>>>>> the ERA "View" UI button (I get the 500 errors copied below).
>>>>>
>>>>> https://era.library.ualberta.ca/items/66cf48a0-f6a7-4429-9349-7507a5475953
>>>>>
>>>>> https://era.library.ualberta.ca/items/34a05341-65e6-4732-a341-134559e8475e
>>>>>
>>>>> Also, the pdfs can be downloaded, but not opened. Is this a known
>>>>> issue? Even though the file sizes are ok and their file extension are ok, I
>>>>> get errors similar to the screenshot below when I try to open them with a
>>>>> pdf reader:
>>>>> <image.png>
>>>>>
>>>>> I noticed the files are retrieved from that old chelsea server, and
>>>>> that the location lists 'proquest', I imagine this is not the right
>>>>> location. Sharon pointed out that these  are part of the digitized theses
>>>>> purchased from proquest back in the day (ingested in batches into ERA).
>>>>> Also from Sharon's email: "We do have copies of these theses on DVDs (and
>>>>> they are likely in Swift as well) so we should be able to recover if
>>>>> needed."
>>>>>
>>>>> Thank you!
>>>>> Mariana
>>>>>
>>>>> 500 Internal Server ErrorError getting
>>>>> http://chelsea.library.ualberta.ca/external/proquest_theses/MM/64/87/5-/MM/94/91/7/MM64875-MM94917.pdf
>>>>> ------------------------------
>>>>>
>>>>> fedora.server.errors.GeneralException: Error getting http://chelsea.library.ualberta.ca/external/proquest_theses/MM/64/87/5-/MM/94/91/7/MM64875-MM94917.pdf
>>>>> 	at fedora.server.storage.DefaultExternalContentManager.get(DefaultExternalContentManager.java:130)
>>>>> 	at fedora.server.storage.DefaultExternalContentManager.getExternalContent(DefaultExternalContentManager.java:203)
>>>>> 	at fedora.server.access.DefaultAccess.getDatastreamDissemination(DefaultAccess.java:1105)
>>>>> 	at fedora.server.access.FedoraAccessServlet.getDatastreamDissemination(FedoraAccessServlet.java:605)
>>>>> 	at fedora.server.access.FedoraAccessServlet.doGet(FedoraAccessServlet.java:458)
>>>>> 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:689)
>>>>> 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:802)
>>>>> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:252)
>>>>> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:173)
>>>>> 	at fedora.server.security.servletfilters.FilterSetup.doFilter(FilterSetup.java:235)
>>>>> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:202)
>>>>> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:173)
>>>>> 	at fedora.server.security.servletfilters.FilterSetup.doFilter(FilterSetup.java:235)
>>>>> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:202)
>>>>> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:173)
>>>>> 	at fedora.server.security.servletfilters.FilterSetup.doFilter(FilterSetup.java:235)
>>>>> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:202)
>>>>> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:173)
>>>>> 	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:213)
>>>>> 	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:178)
>>>>> 	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:524)
>>>>> 	at org.apache.catalina.valves.RequestFilterValve.process(RequestFilterValve.java:282)
>>>>> 	at org.apache.catalina.valves.RemoteAddrValve.invoke(RemoteAddrValve.java:80)
>>>>> 	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:126)
>>>>> 	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:105)
>>>>> 	at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:541)
>>>>> 	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:107)
>>>>> 	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:148)
>>>>> 	at org.apache.coyote.http11.Http11Processor.process(Http11Processor.java:869)
>>>>> 	at org.apache.coyote.http11.Http11BaseProtocol$Http11ConnectionHandler.processConnection(Http11BaseProtocol.java:664)
>>>>> 	at org.apache.tomcat.util.net.PoolTcpEndpoint.processSocket(PoolTcpEndpoint.java:527)
>>>>> 	at org.apache.tomcat.util.net.LeaderFollowerWorkerThread.runIt(LeaderFollowerWorkerThread.java:80)
>>>>> 	at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:684)
>>>>> 	at java.lang.Thread.run(Thread.java:662)
>>>>> Caused by: java.io.IOException: Request failed [404 Not Found]
>>>>> 	at fedora.common.http.WebClient.get(WebClient.java:179)
>>>>> 	at fedora.common.http.WebClient.get(WebClient.java:103)
>>>>> 	at fedora.server.storage.DefaultExternalContentManager.get(DefaultExternalContentManager.java:122)
>>>>> 	... 33 more
>>>>>
>>>>>
>>>>>
>>>>> 500 Internal Server ErrorError getting
>>>>> http://chelsea.library.ualberta.ca/external/proquest_theses/87/10/53/1-/NL/22/92/3/8710531-NL22923.pdf
>>>>> ------------------------------
>>>>>
>>>>> fedora.server.errors.GeneralException: Error getting http://chelsea.library.ualberta.ca/external/proquest_theses/87/10/53/1-/NL/22/92/3/8710531-NL22923.pdf
>>>>> 	at fedora.server.storage.DefaultExternalContentManager.get(DefaultExternalContentManager.java:130)
>>>>> 	at fedora.server.storage.DefaultExternalContentManager.getExternalContent(DefaultExternalContentManager.java:203)
>>>>> 	at fedora.server.access.DefaultAccess.getDatastreamDissemination(DefaultAccess.java:1105)
>>>>> 	at fedora.server.access.FedoraAccessServlet.getDatastreamDissemination(FedoraAccessServlet.java:605)
>>>>> 	at fedora.server.access.FedoraAccessServlet.doGet(FedoraAccessServlet.java:458)
>>>>> 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:689)
>>>>> 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:802)
>>>>> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:252)
>>>>> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:173)
>>>>> 	at fedora.server.security.servletfilters.FilterSetup.doFilter(FilterSetup.java:235)
>>>>> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:202)
>>>>> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:173)
>>>>> 	at fedora.server.security.servletfilters.FilterSetup.doFilter(FilterSetup.java:235)
>>>>> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:202)
>>>>> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:173)
>>>>> 	at fedora.server.security.servletfilters.FilterSetup.doFilter(FilterSetup.java:235)
>>>>> 	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:202)
>>>>> 	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:173)
>>>>> 	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:213)
>>>>> 	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:178)
>>>>> 	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:524)
>>>>> 	at org.apache.catalina.valves.RequestFilterValve.process(RequestFilterValve.java:282)
>>>>> 	at org.apache.catalina.valves.RemoteAddrValve.invoke(RemoteAddrValve.java:80)
>>>>> 	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:126)
>>>>> 	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:105)
>>>>> 	at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:541)
>>>>> 	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:107)
>>>>> 	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:148)
>>>>> 	at org.apache.coyote.http11.Http11Processor.process(Http11Processor.java:869)
>>>>> 	at org.apache.coyote.http11.Http11BaseProtocol$Http11ConnectionHandler.processConnection(Http11BaseProtocol.java:664)
>>>>> 	at org.apache.tomcat.util.net.PoolTcpEndpoint.processSocket(PoolTcpEndpoint.java:527)
>>>>> 	at org.apache.tomcat.util.net.LeaderFollowerWorkerThread.runIt(LeaderFollowerWorkerThread.java:80)
>>>>> 	at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:684)
>>>>> 	at java.lang.Thread.run(Thread.java:662)
>>>>> Caused by: java.io.IOException: Request failed [404 Not Found]
>>>>> 	at fedora.common.http.WebClient.get(WebClient.java:179)
>>>>> 	at fedora.common.http.WebClient.get(WebClient.java:103)
>>>>> 	at fedora.server.storage.DefaultExternalContentManager.get(DefaultExternalContentManager.java:122)
>>>>> 	... 33 more
>>>>>
>>>>>
>>>>> --
>>>>> *Mariana Paredes-Olea*
>>>>> she/her/hers
>>>>> Metadata Specialist
>>>>>
>>>>> University of Alberta Libraries
>>>>> *Amiskwaciwâskahikan / ᐊᒥᐢᑲᐧᒋᕀᐋᐧᐢᑲᐦᐃᑲᐣ / Edmonton*
>>>>> *The University of Alberta is situated on Treaty 6 territory, **homeland
>>>>> of the Métis peoples.*
>>>>>
>>>>>
>>>>>
>>>>
>>>> --
>>>> *Sharon Farnel*
>>>> she/her
>>>> Head, Metadata Strategies
>>>> University of Alberta Library
>>>> sharon.farnel@ualberta.ca | 780-492-3685
>>>>
>>>> *The University of Alberta is situated on traditional Treaty 6
>>>> territory and homeland of the Métis peoples.*
>>>> Amiskwaciwâskahikan / ᐊᒥᐢᑲᐧᒋᕀᐋᐧᐢᑲᐦᐃᑲᐣ / Edmonton
>>>>
>>>
>>>
>>> --
>>>
>>> Matthew Barnett
>>>
>>> Senior Programmer Analyst
>>> University of Alberta Library
>>> 2-10 Cameron Library
>>> Edmonton, Alberta, Canada T6G 2J8
>>> Email: mbarnett@ualberta.ca (he/him)
>>>
>>>
>>
>> --
>> *Sharon Farnel*
>> she/her
>> Head, Metadata Strategies
>> University of Alberta Library
>> sharon.farnel@ualberta.ca | 780-492-3685
>>
>> *The University of Alberta is situated on traditional Treaty 6
>> territory and homeland of the Métis peoples.*
>> Amiskwaciwâskahikan / ᐊᒥᐢᑲᐧᒋᕀᐋᐧᐢᑲᐦᐃᑲᐣ / Edmonton
>>
>
>
> --
> *Sharon Farnel*
> she/her
> Head, Metadata Strategies
> University of Alberta Library
> sharon.farnel@ualberta.ca | 780-492-3685
>
> *The University of Alberta is situated on traditional Treaty 6
> territory and homeland of the Métis peoples.*
> Amiskwaciwâskahikan / ᐊᒥᐢᑲᐧᒋᕀᐋᐧᐢᑲᐦᐃᑲᐣ / Edmonton
>

--

**Comment by mbarnett** [comment]
tl;dr Complicated issue, unclear responsibility, discussion to follow
